vm/devices/user_driver/src/lib.rs

```diff
@@ -63,10 +63,10 @@ pub trait DeviceRegisterIo: Send + Sync {
 /// Device interfaces for DMA.
 pub trait DmaClient: Send + Sync + Inspect {
     /// Allocate a new DMA buffer. This buffer must be zero initialized.
-    ///
-    /// TODO: string tag for allocation?
-    fn allocate_dma_buffer(&self, total_size: usize) -> anyhow::Result<MemoryBlock>;
+    fn allocate_dma_buffer(&self, total_size: usize, tag: String) -> anyhow::Result<MemoryBlock>;
```

Comment:
if you don't mind a bit of churn - changing `tag` to be `debug_tag` would go a long way to clarifying the (current) intent of the tag.

regardless though - there should be some clear comments here and elsewhere pointing out the debug-only nature of these tags (for the time being))